jax/api.py

```diff
@@ -1720,6 +1720,10 @@ def _jvp(fun: lu.WrappedFun, primals, tangents):
                       f"Got primal dtype {_dtype(p)} and so expected tangent dtype "
                       f"{core.primal_dtype_to_tangent_dtype(_dtype(p))}, but got "
                       f"tangent dtype {_dtype(t)} instead.")
+    if np.shape(p) != np.shape(t):
+      raise ValueError("jvp called with inconsistent primal and tangent shapes;"
```

Comment:
One small suggestion: instead of using "inconsistent" we can use the more precise term: "different", because that is what you are actually checking for.

Reply:
Hi,
I have tried to address all the requested changes

Thanks,
Gaurav